tests/install_upgrade_operators/strict_reconciliation/conftest.py

```
@@ -116,12 +119,15 @@ def hco_with_non_default_feature_gates(
 ):
     new_fgs = request.param["fgs"]
-    hco_fgs = hyperconverged_resource_scope_function.instance.to_dict()["spec"]["featureGates"]
+    hco_fgs = hyperconverged_resource_scope_function.instance.to_dict()["spec"].get("featureGates", [])
 
-    for fg in new_fgs:
-        hco_fgs[fg] = True
+    existing_fg_names = {fg["name"] for fg in hco_fgs}
+    updated_fgs = list(hco_fgs)
+    for fg_name in new_fgs:
+        if fg_name not in existing_fg_names:
+            updated_fgs.append({"name": fg_name, "state": FG_STATE_ENABLED})
```

Ruff (0.15.18)

[warning] Use `list.extend` to create a transformed list

(PERF401)